analyzers/tests/SonarAnalyzer.UnitTest/Helpers/TypeDeclarationSyntaxExtensionsTest.cs

```diff
@@ -91,5 +91,148 @@ private void Method2()
+        [DataRow(LanguageVersion.CSharp11)]
+        [DataRow(LanguageVersion.CSharp12)]
+        public void PrimaryConstructor_NoPrimaryConstructor(LanguageVersion languageVersion)
+        {
+            var tree = CSharpSyntaxTree.ParseText("""
+            public class Test
+            {
+            }
+            """, new CSharpParseOptions(languageVersion));
+            var compilation = CSharpCompilation.Create(assemblyName: null, syntaxTrees: new[] { tree });
+            var typeDeclaration = tree.GetCompilationUnitRoot().DescendantNodesAndSelf().OfType<TypeDeclarationSyntax>().Single();
+            typeDeclaration.PrimaryConstructor(compilation.GetSemanticModel(tree)).Should().BeNull();
+        }
+
+        [DataTestMethod]
+        [DataRow(LanguageVersion.CSharp9)]
+        [DataRow(LanguageVersion.CSharp10)]
+        [DataRow(LanguageVersion.CSharp11)]
+        [DataRow(LanguageVersion.CSharp12)]
+        public void PrimaryConstructor_PrimaryConstructorRecord(LanguageVersion languageVersion)
+        {
+            var tree = CSharpSyntaxTree.ParseText("""
+            public record Test(int i)
+            {
+            }
+            """, new CSharpParseOptions(languageVersion));
+            var compilation = CSharpCompilation.Create(assemblyName: null, syntaxTrees: new[] { tree });
+            var typeDeclaration = tree.GetCompilationUnitRoot().DescendantNodesAndSelf().OfType<TypeDeclarationSyntax>().Single();
+            var methodSymbol = typeDeclaration.PrimaryConstructor(compilation.GetSemanticModel(tree));
+            methodSymbol.Should().NotBeNull();
+            methodSymbol.MethodKind.Should().Be(MethodKind.Constructor);
+            methodSymbol.Parameters.Should().ContainSingle();
+            methodSymbol.Parameters.Should().ContainSingle(p => p.Name == "i" && p.Type.SpecialType == SpecialType.System_Int32);
+        }
+
+        [DataTestMethod]
+        [DataRow("class")]
+        [DataRow("struct")]
+        [DataRow("readonly struct")]
+        [DataRow("record struct")]
+#if NET
+        [DataRow("record")]
+        [DataRow("record class")]
+#endif
+        public void PrimaryConstructor_PrimaryConstructorOnClass(string type)
+        {
+            var (tree, model) = TestHelper.CompileCS($$"""
+            {{type}} Test(int i)
+            {
+            }
+            """);
+            var typeDeclaration = tree.GetCompilationUnitRoot().DescendantNodesAndSelf().OfType<TypeDeclarationSyntax>().Single();
+            var methodSymbol = typeDeclaration.PrimaryConstructor(model);
+            methodSymbol.Should().NotBeNull();
+            methodSymbol.MethodKind.Should().Be(MethodKind.Constructor);
+            methodSymbol.Parameters.Should().ContainSingle();
+            methodSymbol.Parameters.Should().ContainSingle(p => p.Name == "i" && p.Type.SpecialType == SpecialType.System_Int32);
+        }
+
+        [TestMethod]
+        public void PrimaryConstructor_EmptyPrimaryConstructor()
```

Comment:
What happens when there's an empty primary constructor and no explicit ctor provided?
```cs
public class Test()
{
}
```

Reply:
The same. I added a test